aptos-move/framework/aptos-framework/sources/reconfiguration_state.move

```diff
@@ -0,0 +1,113 @@
+/// Reconfiguration meta-state resources and util functions.
+module aptos_framework::reconfiguration_state {
+    use std::error;
+    use std::string;
+    use aptos_std::copyable_any;
+    use aptos_std::copyable_any::Any;
+    use aptos_framework::system_addresses;
+    use aptos_framework::timestamp;
+
+    friend aptos_framework::genesis;
+    friend aptos_framework::reconfiguration;
+    friend aptos_framework::reconfiguration_with_dkg;
+    friend aptos_framework::stake;
+
+    const ERECONFIG_NOT_IN_PROGRESS: u64 = 1;
+
+    /// Reconfiguration drivers update this resources to notify other modules of some reconfiguration state.
+    struct State has key {
+        /// The state variant packed as an `Any`.
+        /// Currently the variant type is one of the following.
+        /// - `ReconfigStateInactive`
+        /// - `ReconfigStateActive`
+        variant: Any,
```

Comment:
Why does this need to be a type like StateInactive and StateActive instead of an u64 (standard enum like)?

Reply:
in `ReconfigStateActive` case we need a 2nd field to record when reconfig started. and (to be a bit overthinking) It also may become more complicated in the future, so adopting Any.